code/modules/mining/minerals.dm

```diff
@@ -179,6 +187,9 @@ mineral/cave
 	ore = null
 	UpdateTurf(var/turf/T)
 		if(!istype(T,/turf/unsimulated/floor/asteroid/cave))
-			T.ChangeTurf(/turf/unsimulated/floor/asteroid/cave)
+			if(mined_type == /turf/unsimulated/floor/asteroid) //Because all the cave stuff's made in New(), can't give it mined_type here like gibtonite, instead ya gotta make a new type and reference that.
```

Comment:
Special cases are not special enough to break the rules.

Refactor it if you have to, no hardcoding like this.